### PR TITLE
- Project wipe

### PR DESCRIPTION
**Pull request by malamili**
- QL support - SystemNode
- Better testsuite and loading
- WeaverFile interface
- LocalController

### CODE REVIEW

File: test/WeaverUser.test.coffee [hunk 1]
```diff
@@ -1,9 +1,6 @@
 require("./test-suite")()
```
Comment (carvilsi) [inline test/WeaverUser.test.coffee]: without the end brackets:
require("./test-suite")

File: test/WeaverUser.test.coffee [hunk 2]
```diff
@@ -1,9 +1,6 @@
...
-# Weaver
-Weaver      = require('./../src/Weaver')
-WeaverError = require('./../../weaver-commons-js/src/WeaverError')
-require('./../src/WeaverNode')  # This preloading will be an issue
-require('./../src/WeaverUser')
+
+Weaver = require('./../src/Weaver')
 
 describe 'Weaver User', ->
   this.timeout(2000)
```
Comment (carvilsi) [inline test/WeaverUser.test.coffee]: I think this part is no more necessary:
```before (done) ->
     Weaver.initialize(WEAVER_ADDRESS)
     .then(->
       wipe()
     ).then(->
       done();
     )
     return```

File: test/WeaverFile.test.coffee [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  it 'should create a new file by an upload form', ->
+    fileUploadControl = $("#fileUpload")[0]
+    if fileUploadControl.files.length > 0
+      file = fileUploadControl.files[0]
+      weaverFile = new Weaver.File(file)
+      weaverFile.setFile(file)
+      weaverFile.save()
+
+  # Later this URL will contain some form of session to authorize with
+  it 'should load a file by URL'
```
Comment (carvilsi) [inline test/WeaverFile.test.coffee]: it 'should load a file by URL', ->

File: test/WeaverFile.test.coffee [hunk 4]
```diff
@@ -0,0 +1,33 @@
+require("./test-suite")
+
+describe 'WeaverFile test', ->
+
+  it 'should create a new file by base64', ->
+    base64 = "V2VhdmVyIGlzIEF3ZXNvbWUh"
+    weaverFile = new Weaver.File()
+    weaverFile.setBase64(base64)
+    weaverFile.save()
```
Comment (carvilsi) [inline test/WeaverFile.test.coffee]: Why not just one step, to save it? (Something like `weaverFile.saveBase64(base64)`) 

Is this for temporal files or is required some file persistence, and we'll need to deal with user's names?

weaverFile.setBase64(fileName, base64)

Comment (malamili) [inline test/WeaverFile.test.coffee]: We resolved this

File: test/WeaverUser.test.coffee (hunk 1 above)
Comment (malamili) [inline test/WeaverUser.test.coffee]: Check!

File: test/WeaverUser.test.coffee (hunk 2 above)
Comment (malamili) [inline test/WeaverUser.test.coffee]: Correct!

File: test/WeaverFile.test.coffee (hunk 4 above)
Comment (carvilsi) [inline test/WeaverFile.test.coffee]: 👍

**Pull request merged**